html/modules/custom/reliefweb_utility/src/Controller/ErrorPageController.php

```diff
@@ -0,0 +1,148 @@
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('current_user'),
+      $container->get('request_stack'),
+      $container->get('messenger'),
+      $container->get('typed_data_manager'),
+    );
+  }
+
+  /**
+   * Get the current request.
+   *
+   * @return \Symfony\Component\HttpFoundation\Request
+   *   Current request.
+   */
+  public function getRequest(): Request {
+    return $this->requestStack->getCurrentRequest();
+  }
+
+  /**
+   * Handle the error passed by nginx.
+   */
+  public function handleError() {
+    $request = $this->getRequest();
+
+    // What size was the file the user tried?
+    $size = ByteSizeMarkup::create($request->query->get('size') ?? 0);
+
+    // What is the max size the site allows?
+    $max_size = ByteSizeMarkup::create(Environment::getUploadMaxSize());
+
+    // If this is an ajax callback, we can look up the field.
+    if ($request->isXmlHttpRequest()) {
+
+      $field_name = $request->query->get('field_parents');
+      $field_config = FieldStorageConfig::loadByName('node', $field_name);
+
+      // Instantiate the field, so we can ask about its max size.
+      $field_item_definition = $this->typedDataManager->createDataDefinition('field_item:' . $field_config->getType());
+      $field_item = $this->typedDataManager->createInstance('field_item:' . $field_config->getType(), [
+        'name' => 'wibble',
+        'parent' => NULL,
+        'data_definition' => $field_item_definition,
+      ]);
+      $max_size = ByteSizeMarkup::create($field_item->getMaxFileSize());
+
+      $message = $this->t("The file you are trying to upload is %size, which exceeds the maximum allowed size of %max_size. Please compress the file or choose a smaller one before trying again. If you need help reducing file size or have questions about upload limits, contact support: submit@reliefweb.int",
+        [
+          '%size' => $size,
+          '%max_size' => $max_size,
+        ]
+      );
+
+      $response = [
+        [
+          'selector' => '#edit-' . strtr($field_name, ['_' => '-']),
+          'command'  => 'insert',
+          'method'   => 'append',
+          'data'     => '<div class="form-item--error-message">' . $message . '</div>',
+        ],
+      ];
+      return new JsonResponse($response, 200);
+    }
+
+    // Show an error box, for consistency.
+    $this->messenger->addError($this->t("File Too Large"));
+
+    $message = $this->t("The file you are trying to upload is %size, which exceeds the maximum allowed size of %max_size. Please compress the file or choose a smaller one before trying again. If you need help reducing file size or have questions about upload limits, contact support: submit@reliefweb.int",
+      [
+        '%size' => $size,
+        '%max_size' => $max_size,
+      ]
+    );
+
+    return [
+      '#markup' => $message,
+    ];
```

Comment:
```suggestion
    return [
      '#type' => 'inline_template',
      '#template' => '<p class="rw-error-413">{{ message }}</p>',
      '#context' => ['message' => $message],
    ];
```

Just for minimum styling with some spacing between the title and message.